src/VecSim/index_factories/svs_factory.cpp

```diff
@@ -201,6 +201,16 @@ VecSimIndex *NewIndex(const VecSimParams *params, bool is_normalized) {
     return NewIndexImpl(params, is_normalized);
 }
 
+#if BUILD_TESTS
+VecSimIndex *NewIndex(const std::string &location, const VecSimParams *params, bool is_normalized) {
+    auto index = NewIndexImpl(params, is_normalized);
+    if (index != nullptr) {
+        index->loadIndex(location);
```

Comment:
It seems like this construction forced you to declare `loadIndex()` in `VecSimIndexInterface`.
For VectorSimilarity project maintainability,
I would recommend to declare `loadindex()` in `Serializer` (next to `saveIndex()`) and use 'side-cast' here by `dynamic_cast<>()` expression:
```suggestion
        auto serializer = dynamic_cast<Serializer*>(index);
        assert(serializer != nullptr && "VecSimIndex->Serializer side-casting failed - check SVSIndex inheritance");
        serializer->loadIndex(location);
```